### PR TITLE
Remove explicit print in lightning-net-tokio, reduce redundant block connection logging

### LINKED ISSUE #980: Clean up block connection logging in ChainMonitor/ChannelMonitor
We should improve the logging to be less redundant and increase coverage of what's going on, especially at debug level. See https://github.com/rust-bitcoin/rust-lightning/pull/970#discussion_r661107957

### PR DESCRIPTION
**Pull request by TheBlueMatt**
Fixes #980.

### CODE REVIEW

File: lightning/src/ln/peer_handler.rs [hunk 1]
```diff
@@ -1344,6 +1345,9 @@ impl<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, L: Deref> PeerManager<D
 			Some(peer) => {
 				match peer.their_node_id {
 					Some(node_id) => {
+						log_trace!(self.logger,
+							"Handling peer disconnection of {}, with {}future connection to the peer possible.",
+							node_id, if no_connection_possible { "no " } else { "" });
```
Comment (jkczyz) [inline lightning/src/ln/peer_handler.rs:1350]: Should this use `log_pubkey!`?

File: lightning/src/chain/chainmonitor.rs [hunk 2]
```diff
@@ -170,6 +171,7 @@ where
 
 	fn block_disconnected(&self, header: &BlockHeader, height: u32) {
 		let monitors = self.monitors.read().unwrap();
+		log_debug!(self.logger, "Latest block disconnected, new best block {} at height {} provided via block_disconnected", header.block_hash(), height);
```
Comment (jkczyz) [inline lightning/src/chain/chainmonitor.rs:174]: I believe this should be the header and height of the disconnected block, not the new best block. We decrement the height and get the previous block hash in `ChannelMonitor::block_disconnected`.

File: lightning/src/ln/peer_handler.rs [hunk 3]
```diff
@@ -667,6 +667,7 @@ impl<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, L: Deref> PeerManager<D
 		match self.do_read_event(peer_descriptor, data) {
 			Ok(res) => Ok(res),
 			Err(e) => {
+				log_trace!(self, "Peer sent invalid data or we decided to disconnect due to a protocol error");
```
Comment (jkczyz) [inline lightning/src/ln/peer_handler.rs:670]: `self.logger`

File: lightning/src/chain/chainmonitor.rs [hunk 4]
```diff
@@ -170,6 +171,7 @@ where
 
 	fn block_disconnected(&self, header: &BlockHeader, height: u32) {
 		let monitors = self.monitors.read().unwrap();
+		log_debug!(self.logger, "Latest block disconnected, block {} at height {} disconnected via block_disconnected", header.block_hash(), height);
```
Comment (jkczyz) [inline lightning/src/chain/chainmonitor.rs:174]: To remove some redundancy with the word "disconnected" and for consistency with `block_connected`, how about `"Latest block {} at height {} removed via block_disconnected"`?

File: lightning/src/chain/chainmonitor.rs [hunk 5]
```diff
@@ -187,20 +189,23 @@ where
 	P::Target: channelmonitor::Persist<ChannelSigner>,
 {
 	fn transactions_confirmed(&self, header: &BlockHeader, txdata: &TransactionData, height: u32) {
+		log_debug!(self.logger, "{} transactions confirmed at height {} in block {}", txdata.len(), height, header.block_hash());
```
Comment (jkczyz) [inline lightning/src/chain/chainmonitor.rs:192]: Maybe qualify "transactions" with "relevant" so it's clear the block may have had more transactions? Though I suppose the user could pass anything to this so, they may not all be relevant.

File: lightning/src/chain/channelmonitor.rs [hunk 6]
```diff
@@ -2044,6 +2041,7 @@ impl<Signer: Sign> ChannelMonitorImpl<Signer> {
 		F::Target: FeeEstimator,
 		L::Target: Logger,
 	{
+		log_trace!(logger, "Processing {} transactions for block at height {}.", txn_matched.len(), conf_height);
```
Comment (jkczyz) [inline lightning/src/chain/channelmonitor.rs:2044]: Similarly, perhaps "matched transactions"?

**Review by jkczyz — APPROVED** [review]

**Comment by TheBlueMatt** [comment]
Addressed feedback and added one more commit to clean up docs.

**Review by jkczyz — APPROVED** [review]

File: lightning/src/chain/chainmonitor.rs [hunk 7]
```diff
@@ -187,20 +189,23 @@ where
...
 	fn transactions_confirmed(&self, header: &BlockHeader, txdata: &TransactionData, height: u32) {
+		log_debug!(self.logger, "{} provided transactions confirmed at height {} in block {}", txdata.len(), height, header.block_hash());
 		self.process_chain_data(header, txdata, |monitor, txdata| {
 			monitor.transactions_confirmed(
 				header, txdata, height, &*self.broadcaster, &*self.fee_estimator, &*self.logger)
 		});
 	}
 
 	fn transaction_unconfirmed(&self, txid: &Txid) {
+		log_debug!(self.logger, "{} reorganized out of chain", txid);
```
Comment (valentinewallace) [inline lightning/src/chain/chainmonitor.rs:200]: suggestion: s/{}/txid {}

**Review by valentinewallace — APPROVED** [review]

File: lightning/src/chain/chainmonitor.rs (hunk 7 above)
Comment (TheBlueMatt) [reply lightning/src/chain/chainmonitor.rs:200]: How about `Transaction {}`?

**Comment by TheBlueMatt** [comment]
Pushed a trivial change as suggested by val and squashed, diff from Jeff's ACK:

```
$ git diff-tree -U1 77178c6 5d847048
diff --git a/lightning/src/chain/chainmonitor.rs b/lightning/src/chain/chainmonitor.rs
index 8100153a..8969427a 100644
--- a/lightning/src/chain/chainmonitor.rs
+++ b/lightning/src/chain/chainmonitor.rs
@@ -199,3 +199,3 @@ where
 	fn transaction_unconfirmed(&self, txid: &Txid) {
-		log_debug!(self.logger, "{} reorganized out of chain", txid);
+		log_debug!(self.logger, "Transaction {} reorganized out of chain", txid);
 		let monitors = self.monitors.read().unwrap();
```

**Review by valentinewallace — APPROVED** [review]

**Comment by TheBlueMatt** [comment]
Oops, had to tweak the last commit cause I misunderstood doclinks, diff is trivial, so will land after CI:
```
$ git diff-tree -U1 5d847048 35d0b7aa
diff --git a/lightning/src/ln/script.rs b/lightning/src/ln/script.rs
index bee544ea..4e81d76a 100644
--- a/lightning/src/ln/script.rs
+++ b/lightning/src/ln/script.rs
@@ -90,3 +90,4 @@ impl ShutdownScript {
 	///
-	/// Note for version-zero witness scripts you must use [`new_p2wpkh`] or [`new_p2wsh`] instead.
+	/// Note for version-zero witness scripts you must use [`ShutdownScript::new_p2wpkh`] or
+	/// [`ShutdownScript::new_p2wsh`] instead.
 	///
$
```

**Pull request merged**